subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/auth2/BackgroundTokenRefresh.kt

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.subscriptions.impl.auth2
+
+import android.content.Context
+import androidx.work.BackoffPolicy
+import androidx.work.Constraints
+import androidx.work.CoroutineWorker
+import androidx.work.ExistingPeriodicWorkPolicy
+import androidx.work.NetworkType
+import androidx.work.PeriodicWorkRequestBuilder
+import androidx.work.WorkManager
+import androidx.work.WorkerParameters
+import com.duckduckgo.anvil.annotations.ContributesWorker
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.subscriptions.impl.AccessTokenResult.Success
+import com.duckduckgo.subscriptions.impl.SubscriptionsManager
+import com.duckduckgo.subscriptions.impl.auth2.TokenRefreshWorker.Companion.REFRESH_WORKER_NAME
+import com.squareup.anvil.annotations.ContributesBinding
+import java.time.Duration
+import java.util.concurrent.TimeUnit.MINUTES
+import javax.inject.Inject
+
+/**
+ * The refresh token is valid for 1 month. We should ensure it doesn’t expire, as that could result in the user being signed out.
+ * In practice, if the app is actively used, tokens will be refreshed more frequently since the access token is only valid for 4 hours.
+ * Otherwise, this worker ensures that we obtain a new refresh token before the current one expires.
+ */
+interface BackgroundTokenRefresh {
+    fun schedule()
+}
+
+@ContributesBinding(AppScope::class)
+class BackgroundTokenRefreshImpl @Inject constructor(
+    val workManager: WorkManager,
+) : BackgroundTokenRefresh {
+    override fun schedule() {
+        val workRequest = PeriodicWorkRequestBuilder<TokenRefreshWorker>(repeatInterval = Duration.ofDays(7))
```

Comment:
Why are we scheduling this work to repeat every 7 days? Access token expires every 4 hours and refresh tokens expire every month?